### ISSUE TITLE
A11y - Search box close butting inaccessible using keyboard

### ISSUE THREAD

**Issue by joshblack** [carbon-design-system/carbon#730]
@ericabr commented on [Mon Apr 23 2018](https://github.com/carbon-design-system/carbon-components-react/issues/830)

## A11y for search box
A11y manual check fails keyboard navigation: Using the search box in our UI doesn't pass a11y tests because the icon used to clear our the contents can't be reached via the keyboard.

## Detailed description
A person using the keyboard cannot use the "x" to clear the contents of the box. The user will have to leave focus of the item, return the focus and delete the contents manually.
![focus_search_close_btn](https://user-images.githubusercontent.com/33731164/39159516-f08086ee-4733-11e8-9733-5df2ab319ca1.gif)

## Steps to reproduce the issue
This can be reproduced in the storybook

---

@joshblack commented on [Mon Apr 23 2018](https://github.com/carbon-design-system/carbon-components-react/issues/830#issuecomment-383785218)

Hi @ericabr! 👋 Thanks so much for taking the time to make this issue! Definitely is an a11y violation, we should probably be coding the clear icon up as a button to help remedy this 🙂 Will gladly accept any PRs that help fix this!
